### DISCUSSION TITLE
Generalizing the build system

### DISCUSSION THREAD

**Discussion by h-milz** [lvgl-micropython/lvgl_micropython#220] [Ideas]
Your build system is a real gem. I have yet to find another out-of-the-box working build system for MicroPython in general, and more often than not, people fail building generic Micropython images for their boards. As opposed to Circuitpython, afaik there is no automated build system for generic Micropython images anywhere, and in various fora, people keep asking if someone could not build an image for this or that board for them. 

Given that many people run their Micropython boards without graphics (in real-world IoT scenarios) I was thinking if one could not use your well-functioning build infrastructure also for building images without graphics. I understand that this would sort of defeat the original purpose of what you are doing but it would open another world of image building for many users. 

All this would require as far as I can tell is a command-line option doing the trick.

**Comment by kdschlosser** [comment]
Yes it can be used to compile a vanilla Micropython. It wouldn't be all that difficult to remove the LVGL specific parts. The build script does more than just compiling. It also makes changes to some of the MicroPython port files to make things function better. Stuff like being able to run threads on both cores of the ESP32. Another is allowing the SDCard to share an SPI bus with other devices, which is not something that MicroPython is able to do.

IDK if you know this or not you can also start the build using makefile as well. The syntax is a slight bit different, build arguments that starts with `--` need to have the `--` removed. I think that is the only difference.
